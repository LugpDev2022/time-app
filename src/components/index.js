export * from './AppNavbar';
export * from './NavbarBrand';
export * from './NavbarLink';
export * from './styled';
