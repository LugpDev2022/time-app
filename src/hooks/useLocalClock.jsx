import { useState, useEffect } from 'react';

export const useLocalClock = () => {
  const [{ hours, minutes }, setTime] = useState({
    hours: new Date().getHours().toString(),
    minutes: new Date().getMinutes().toString(),
  });

  useEffect(() => {
    if (hours.length < 2) {
      setTime({ minutes, hours: `0${hours}` });
    }

    if (minutes.length < 2) {
      setTime({ minutes: `0${minutes}`, hours });
    }
  }, []);

  useEffect(() => {
    setInterval(() => {
      const date = new Date();
      const newMinutes = date.getMinutes().toString();
      const newHours = date.getHours().toString();

      if (minutes !== newMinutes && newMinutes.length < 2) {
        if (newHours.length < 2) {
          setTime({ hours: `0${newHours}`, minutes: `0${newMinutes}` });
          return;
        }

        setTime({ hours: newHours, minutes: `0${newMinutes}` });
        return;
      }

      if (minutes !== newMinutes) {
        if (newHours.length < 2) {
          setTime({ hours: `0${newHours}`, minutes: `0${newMinutes}` });
          return;
        }

        setTime({ hours: newHours, minutes: newMinutes });
        return;
      }
    }, 5000);
  }, []);

  return { hours, minutes };
};
