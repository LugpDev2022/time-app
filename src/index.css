body {
  background-color: #1b1e2d;
  color: #4cb4ee;
}
