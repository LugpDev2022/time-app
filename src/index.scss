@import 'bootstrap/scss/bootstrap';

body {
  background-color: #1b1e2d !important;
}
