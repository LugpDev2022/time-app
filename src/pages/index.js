export * from './clock/ClockPage';
export * from './timer/TimerPage';
export * from './worldClock/WorldClockPage';
