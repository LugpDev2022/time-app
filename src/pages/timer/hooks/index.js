export * from "./useTimer";
export * from "./useNewTimer";
