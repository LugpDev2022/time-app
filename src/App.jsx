import React from "react";
import { Router } from "./router";
import { AppNavbar } from "./components";

const App = () => (
  <>
    <AppNavbar />

    <Router />
  </>
);

export default App;
