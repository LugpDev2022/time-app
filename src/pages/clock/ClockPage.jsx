import { Col, Container, Row } from 'react-bootstrap';
import { Box, BigNumber, Title } from '../../components';
import { useLocalClock } from './hooks';

export const ClockPage = () => {
  const { hours, minutes } = useLocalClock();

  return (
    <Box>
      <Container className='h-100 d-flex flex-column justify-content-center'>
        <Row>
          <Col>
            <Title>Local Clock</Title>
          </Col>
        </Row>

        <Row>
          <div className='d-flex justify-content-center'>
            <BigNumber>{`${hours}:${minutes}`}</BigNumber>
          </div>
        </Row>
      </Container>
    </Box>
  );
};
