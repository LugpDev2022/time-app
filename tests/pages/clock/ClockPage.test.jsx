import { render, screen } from '@testing-library/react';
import { ClockPage } from '../../../src/pages';
import { useLocalClock } from '../../../src/pages/clock/hooks';

jest.mock('../../../src/pages/clock/hooks/useLocalClock');

describe('tests on <ClockPage />', () => {
  test('should match snapshot', () => {
    useLocalClock.mockReturnValue({
      hours: '05',
      minutes: '07',
    });

    const { container } = render(<ClockPage />);
    expect(container).toMatchSnapshot();
  });

  test('should show correctly the clock', () => {
    useLocalClock.mockReturnValue({
      hours: '05',
      minutes: '07',
    });

    render(<ClockPage />);
    expect(screen.getByText('05:07'));
  });
});
