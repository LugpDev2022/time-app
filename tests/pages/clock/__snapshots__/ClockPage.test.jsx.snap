// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <ClockPage /> should match snapshot 1`] = `
<div>
  <div
    class="sc-gswNZR pUmhY"
  >
    <div
      class="h-100 d-flex flex-column justify-content-center container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h1
            class="sc-dkrFOg dcibta"
          >
            Local Clock
          </h1>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="d-flex justify-content-center"
        >
          <span
            class="sc-bcXHqe irHAIw"
          >
            05:07
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
